File: src/adapter/festa/fantasia/RoupaDeCaveira.java
package adapter.festa.fantasia;

// Adaptador especializado

public class RoupaDeCaveira extends Fantasia {

    public RoupaDeCaveira(Pessoa pessoa) {
        super(pessoa);
    }

    @Override
    public String nomeDoPersonagem() {
        return "Uma Caveira com a voz do " + pessoa.obterNome();
    }

    @Override
    public String descricaoDaRoupa() {
        return "Roupa preta inteira com desenho dos ossos";
    }
}

File: src/adapter/festa/fantasia/RoupaDeCaveira.kt
package adapter.festa.fantasia

// Adaptador especializado

class RoupaDeCaveira(pessoa: Pessoa) : Fantasia(pessoa) {

    override fun nomeDoPersonagem(): String {
        return "Uma Caveira com a voz do " + pessoa.obterNome()
    }

    override fun descricaoDaRoupa(): String {
        return "Aparencia preta inteira com desenho dos ossos"
    }
}
